### PR TITLE
fix(challenges): Update challenge text and assertion test

### PR DESCRIPTION
**Pull request by lasjorg**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

1. Change text from calling the variable a private property, to just a private variable. The variable is not a property.

2. Change the assertion test to check for not undefined on the return value instead of the hardcoded check for the number 15. That value is not specified in the text, it also makes the test pass by just returning the number 15 instead of the value of the variable.

3. I'm having some text formatting issues i have to add line breaks to get the text to wrap in paragraphs. Not sure why?

I'm not that comfortable with Chai or the FCC testing to make a PR without some guidance. The simplest test i came up with is just testing for !== undefined, but it might be better to test for a Truthy value.

Please advise if there is a better way to test this and if i have done this correctly. I have tested this locally by running the challenge.

Side note: The name of this challenge is also something i would suggest be changed but i have not come up with a better name yet.

Closes #34785

### CODE REVIEW

**Review by ezioda004 — COMMENTED** [review]
I like the change of private property to private variable as you stated its not really a private property.

Regarding updating the tests. I'm not sure why you'd check if the value its returning isnt `undefined`.
The function could just return any value and that'd pass the test and would essentially defeat the purpose of the challenge.

**Comment by lasjorg** [comment]
@ezioda004 Well the idea was to check that the variable was actually assigned something and that the variable is what gets returned. I don't how returning the number 15 is relevant to the point of the challenge.

As it is now, you can pass the test like this:

```
function Bird() {
  this.getWeight = function() {
    return 15;
  }
}
```

The test does not check for the existence of the variable, nor that the variable is what is getting returned.


BTW. This is the solution to the challenge:

```
function Bird() {
  let weight = 15;
  
  this.getWeight = function() {
    return weight;
  }
}
```

I'm sure there is a better way to do the test then what I did, which is why I asked if anyone had a better way to test it.

**Comment by lasjorg** [comment]
@ezioda004 I'm not too hot at regex, if you have the time can you tell me if this would be an acceptable solution for the test.
```
  - text: The <code>weight</code> property should be a private variable.
    testString: assert(code.match(/(var|let|const)\s*(weight)\s*\=\s*\d+\;/g), 'The <code>weight</code> property should be a private variable.');
  - text: Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.
    testString: assert((new Bird()).getWeight() !== undefined, 'Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.');
  - text: Your code should return the private variable <code>weight</code>.
    testString: assert(code.match(/(return)\s*(weight)\;/g), 'Your code should return the private variable <code>weight</code>.');
```
Or I can make it check for the number 15 and also inform the user that the variable should have that value.
```
  - text: The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.
    testString: assert(code.match(/(var|let|const)\s*(weight)\s*\=\s*(15)\;/g), 'The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.');
  - text: Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.
    testString: assert((new Bird()).getWeight() === 15, 'Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.');
  - text: Your code should return the private variable <code>weight</code>.
    testString: assert(code.match(/(return)\s*(weight)\;/g), 'Your code should return the private variable <code>weight</code>.');
```
Or I can also just revert the test back if you don't think its worth it.

**Comment by lasjorg** [comment]
Well, I just realized that the last regex won't work with ES6 syntax. Here is a revised version of the last assert.

`assert(code.match(/(return)\s*(weight)\;/g) || code.match(/\(\s*\)\s*\=\>\s*(weight)\;/g)`

To allow for this syntax

`this.getWeight = () => weight;`

**Comment by ezioda004** [comment]
Hi @lasjorg,

Sorry, I cant check the PR yet, but will try to do over the weekend.
Meanwhile I'll ask other mods to check the PR.

cc: @RandellDawson, @scissorsneedfoodtoo, @cmccormack, @moT01

**Comment by RandellDawson** [comment]
@lasjorg You will need to commit the changes you mention above, so we can test the PR locally.

**Comment by lasjorg** [comment]
@RandellDawson I was just hoping for some feedback first.

The branch name is not accurate now, as it is not testing for "not undefined" anymore.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md [hunk 1]
```diff
@@ -24,10 +28,12 @@ Change how <code>weight</code> is declared in the <code>Bird</code> function so
...
-    testString: assert((new Bird()).getWeight() === 15, 'Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the <code>weight</code>.');
+  - text: The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.
+    testString: assert(code.match(/(var|let|const)\s+(weight)\s*\=\s*(15)\;/g), 'The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.');
+  - text: Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.
+    testString: assert((new Bird()).getWeight() === 15, 'Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the value of the private variable <code>weight</code>.');
+  - text: Your <code>getWeight</code> function should return the private variable <code>weight</code>.
+    testString: assert(code.match(/(return)\s+(weight)\;/g) || code.match(/\(\s*\)\s*\=\>\s*(weight)\;/g), 'Your <code>getWeight</code> function should return the private variable <code>weight</code>.');
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md]: Try this more compact version:
```suggestion
    testString: assert(code.match(/(return\s+)|(\(\s*\)\s*\=\>\s*)weight\;?/g), 'Your <code>getWeight</code> function should return the private variable <code>weight</code>.');
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md [hunk 2]
```diff
@@ -24,10 +28,12 @@ Change how <code>weight</code> is declared in the <code>Bird</code> function so
 
 ```yml
 tests:
-  - text: The <code>weight</code> property should be a private variable.
-    testString: assert(!code.match(/this\.weight/g), 'The <code>weight</code> property should be a private variable.');
-  - text: Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the <code>weight</code>.
-    testString: assert((new Bird()).getWeight() === 15, 'Your code should create a method in <code>Bird</code> called <code>getWeight</code> that returns the <code>weight</code>.');
+  - text: The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.
+    testString: assert(code.match(/(var|let|const)\s+(weight)\s*\=\s*(15)\;/g), 'The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.');
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md]: The following is a simpler version and allows no semi-colons on end:
```suggestion
    testString: assert(code.match(/(var|let|const)\s+weight\s*\=\s*15\;?/g), 'The <code>weight</code> property should be a private variable and should be assigned the value of <code>15</code>.');
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md [hunk 3]
```diff
@@ -6,11 +6,15 @@ challengeType: 1
...
-<blockquote>function Bird() {<br>&nbsp;&nbsp;let hatchedEgg = 10; // private property<br><br>&nbsp;&nbsp;this.getHatchedEggCount = function() { // publicly available method that a bird object can use<br>&nbsp;&nbsp;&nbsp;&nbsp;return hatchedEgg;<br>&nbsp;&nbsp;};<br>}<br>let ducky = new Bird();<br>ducky.getHatchedEggCount(); // returns 10</blockquote>
+
+The simplest way to make this public property private is by creating a variable within the constructor function. This changes the scope of that variable to be within the constructor function versus available globally. This way, the variable can only be accessed and changed by methods also within the constructor function.
+<blockquote>function Bird() {<br>&nbsp;&nbsp;let hatchedEgg = 10; // private variable<br><br>&nbsp;&nbsp;this.getHatchedEggCount = function() { // publicly available method that a bird object can use<br>&nbsp;&nbsp;&nbsp;&nbsp;return hatchedEgg;<br>&nbsp;&nbsp;};<br>}<br>let ducky = new Bird();<br>ducky.getHatchedEggCount(); // returns 10</blockquote>
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md]: I think the inline comment makes the line too long.  Can you try:
```suggestion
<blockquote>function Bird() {<br>&nbsp;&nbsp;let hatchedEgg = 10; // private variable<br><br>&nbsp;&nbsp;/* publicly available method that a bird object can use */<br>&nbsp;&nbsp;this.getHatchedEggCount = function() { <br>&nbsp;&nbsp;&nbsp;&nbsp;return hatchedEgg;<br>&nbsp;&nbsp;};<br>}<br>let ducky = new Bird();<br>ducky.getHatchedEggCount(); // returns 10</blockquote>
```

**Review by RandellDawson — CHANGES_REQUESTED** [review]
See my proposed changes to clean up the long inline comment and simplify the regex a bit more for two of the tests.  Other than that, I think the changes of this PR look great.

**Comment by RandellDawson** [comment]
@lasjorg You can always rename your branch to something more suitable.

**Comment by lasjorg** [comment]
Thank you for taking the time to help with this. I have made all the suggested changes and tested it locally, everything looks good to me.

I was going to ask about the renaming before I saw your comment, I will rename it.

I don't think this matters but I noticed that using an unassigned const, makes the last test fail. I'm not sure I understand why. Is it not just doing string matching? Why would it matter if it is an unassigned const vs an unassigned let or var. I know an unassigned const is not valid syntax I'm just not sure how that is affecting the match() method. Anyway, I guess it doesn't matter really I was just curious.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md (hunk 2 above)
Comment (lasjorg) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/use-closure-to-protect-properties-within-an-object-from-being-modified-externally.english.md]: I thought we intentionally tested for semicolons, is that not the case? Personally, I'm fine with not testing for them, I just want to make sure it is inline with the curriculum.

**Comment by lasjorg** [comment]
@RandellDawson Hmm not sure what happened? I was afraid I would mess something up renaming the branch.

I ran these commands:

```
git branch -m fix/remove-property-references-and-test-for-undefined-return-value fix/remove-property-references-and-update-tests
git push origin :fix/remove-property-references-and-test-for-undefined-return-value
git push --set-upstream origin fix/remove-property-references-and-update-tests
```

**Comment by RandellDawson** [comment]
Did you close the PR manually?

JavaScript does not allow the use of `const` without assign a value to the variable.

**Comment by RandellDawson** [comment]
Maybe Github automatically close the PR, because the branch is non-existent? Anyway,  just create a new PR with your renamed branch.

**Comment by lasjorg** [comment]
> @lasjorg Did you close the PR manually?

No I have no idea what happened (well other then the renaming).

> JavaScript does not allow the use of `const` without assigning a value to the variable.

I know, my quistion was why does it affect the match() method? Is it because of how the `code` variable looks with invalid JS in it.

**Comment by RandellDawson** [comment]
@lasjorg I am testing your forked branch right now, so I can try and answer your question.

**Comment by RandellDawson** [comment]
Let me know when you create the new PR.  You should be able to copy and paste the PR body from the one that self-closed.

**Comment by lasjorg** [comment]
@RandellDawson Can I not just go to my repo and select the new branch, then click pull request.
Although, when I do that it tells me "Can’t automatically merge. Don’t worry, you can still create the pull request.". Is that a problem?

**Comment by RandellDawson** [comment]
That should work.  There may be a merge conflict now.  Once you create the PR, we will deal with it if one appears.

**Pull request closed without merging**